examples/react/start-basic-apollo-client/app/apollo/index.ts

```diff
@@ -0,0 +1,36 @@
+import { defaultTransformer } from '@tanstack/react-router'
+import { createTransportedQueryPreloader } from './createQueryPreloader'
+import { getApolloTransformer } from './getApolloTransformer'
+import { ApolloProvider } from './ApolloProvider'
+import type { PreloadQueryFunction } from '@apollo/client'
+import type { AnyRouter } from '@tanstack/react-router'
+import type { ApolloClient } from './ApolloClient'
+
+export function routerWithApolloClient<TRouter extends AnyRouter>(
+  router: TRouter['options']['context'] extends {
+    apolloClient: ApolloClient
+    preloadQuery: PreloadQueryFunction
+  }
+    ? TRouter
+    : never,
+  apolloClient: ApolloClient,
+): TRouter {
+  // @ts-ignore unavoidable due to the ternary in arguments
+  router.options.context.apolloClient = apolloClient
+
+  // it would be nice to do this in the long run
+  // @ts-ignore unavoidable due to the ternary in arguments
+  //   router.options.context.preloadQuery = router.isServer
+  //     ? createTransportedQueryPreloader(apolloClient)
+  //     : createQueryPreloader(apolloClient)
+
+  // @ts-ignore unavoidable due to the ternary in arguments
+  router.options.context.preloadQuery =
+    createTransportedQueryPreloader(apolloClient)
+  router.options.transformer = defaultTransformer.withTransformers([
```

Comment:
I'm gonna be honest, I would prefer to have a `router.options.transformers` option - that way we could merge user-supplied transformers with "library-supplied" ones, which is currently not possible.